security/advancedtls/advancedtls.go

```diff
@@ -34,7 +34,10 @@ import (
 )
 
 // VerificationFuncParams contains the parameters available to users when implementing CustomVerificationFunc.
+// The fields contain in this struct are used when making custom verification decisions, and are hence read-only.
+// Setting these fields wouldn't have any effect.
 type VerificationFuncParams struct {
+	// This field is only meaningful for client side. On server side, this field would be an empty string.
 	ServerName     string
 	RawCerts       [][]byte
 	VerifiedChains [][]*x509.Certificate
```

Comment:
Optional: Although these aren't changed in this PR, it might be useful to have doc strings for all the fields.